src/wasm/wasm.cpp

```diff
@@ -1556,10 +1569,16 @@ void StackSwitch::finalize() {
   if (handleUnreachableOperands(this)) {
     return;
   }
+  if (cont->type.isNull()) {
+    // This will never be executed and the instruction will not be emitted.
+    // Model this with an uninhabitable cast type.
+    type = cont->type.with(NonNullable);
```

Comment:
Same problem, although `StackSwitch` does return a continuation. It just might return other values as well.